.github/workflows/pr.yaml

```diff
@@ -0,0 +1,26 @@
+# This is a template `pr.yaml` file for snaps
+# This file is managed by bootstack-charms-spec and should not be modified
+# within individual snap repos. https://launchpad.net/bootstack-charms-spec
+
+name: PR workflow running lint checkers, unit and functional tests
+
+on:
+  push:
+    branches: [ master, main ]
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+    branches: [ master, main ]
+  pull_request_review:
+    types: [ submitted ]
+
+jobs:
+  pr:
+    uses: canonical/bootstack-actions/.github/workflows/pull-request.yaml@main
```

Comment:
Non-blocking question: why don' t we create an `action.yaml` file in bootstack-actions repo with what's currently in `.github/workflows/pull-request.yaml`? This why we can simply have `uses: canoncail/bootstack-actions` here

Reply:
Workflows needs to be specifically in `.github/workflows` folders.

Reply:
Actually, if you look at other custom actions, e.g. [action/checkout](https://github.com/actions/checkout) or [canonical/setup-lxd](https://github.com/canonical/setup-lxd), they all have `action.yaml` in the root dir indicating that this is a custom action repo. Here is the [documentation](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action#creating-an-action-metadata-file) for creating a composite action and I think we should follow it

Reply:
Yes, but those are github actions and not re-usable workflows. I'll check it again.

Reply:
Ah I see what you mean. If you want them to be separate jobs instead of a flat list of steps, we need to use re-usable workflows instead of composite actions. And yes, for re-usable workflows, we need to define them in `.github/workflows`.